client/src/mvc/tool/tool-form.js

```diff
@@ -49,6 +50,25 @@ const View = Backbone.View.extend({
                                     window.location = getAppRoot();
                                     return;
                                 }
+
+                                const client_version = options.version || options.tool_id;
+
+                                if (
+                                    data.versions &&
+                                    client_version &&
+                                    !data.versions.includes(client_version.replace(/.*\//, ""))
+                                ) {
+                                    Toast.warning(
```

Comment:
Instead of this Toast block you may also set: `data.message = "Requested version unavailable.";`. This will then display a warning in the form's `alert` element.

Reply:
@guerler 
that's a very very good advise. thanks!